Provide --sclize option

irushchyshyn opened the pull request:
Enable generating sclized spec files using spec2scl.
The following spec2scl options can be provided alongside:

- --no-meta-runtime-dep
- --no-meta-buildtime-dep
- --skip-functions
- --no-deps-convert
- --list-file

Issue #100

irushchyshyn committed:
Provide --sclize option

Enable generating sclized spec files using spec2scl.
The following spec2scl options can be provided alongside:
--no-meta-runtime-dep
--no-meta-buildtime-dep
--skip-functions
--no-deps-convert
--list-file

mcyprian commented on pyp2rpm/bin.py:
```diff
@@ -8,6 +8,10 @@
 import click
+try:
+    from spec2scl.convertor import Convertor as SclConvertor
+except:
```
Can something except ImportError (ModuleNotFoundError) occur here? Maybe we could log the information when something unexpected happens.

irushchyshyn replied:
Added an extra check in f83a092

mcyprian commented on pyp2rpm/bin.py:
```diff
@@ -98,6 +126,11 @@ def main(package, v, d, s, r, proxy, srpm, p, b, o, t, venv):
+    if sclize:
+        converted = convert_to_scl(converted, scl_kwargs)
+        if not converted:
+            return
```
I caught the idea of this part, but at the beginning it was not clear to me in which cases is the return value of convert_to_scl None/Empty string. Can you add a short comment please.

irushchyshyn replied:
I reworked the validation a bit, so this is not needed anymore.

mcyprian reviewed (changes requested):
Simple and elegant solution, a few things could be improved in my opinion:
- when running pyp2rpm package --no-meta-runtime-dep it seems this option is simply ignored. Combining these options with --sclize should be required or user should be warn that option has no effect in this case.
- I know that help is autogenerated in click, but isn't it possible to make the fact that the last 4 options are SCL specific a bit more obvious in the help message?
- pyp2rpm's man page (file pyp2rpm.1 in the main project directory) should be updated as well. It should be possible to generate man page from the help message directly, but I've never tried how it works https://www.gnu.org/software/help2man/

irushchyshyn committed:
Review updates

- warn that --sclize option is required when SCL related options are
supplied;
- make SCL related options more obvious on help message (in a separate
group);
- handle non import related errors when importing SclConvertor;
- update man file.

irushchyshyn commented:
Thank you for the review @mcyprian ! 

>     when running pyp2rpm package --no-meta-runtime-dep it seems this option is simply ignored. Combining these options with --sclize should be required or user should be warn that option has no effect in this case.
I have added a validation for a case like that.

>     I know that help is autogenerated in click, but isn't it possible to make the fact that the last 4 options are SCL specific a bit more obvious in the help message?
I have modified help formatting for options, to put SCL related options in a separate options group. Please let me know what you think and how it looks now.

    pyp2rpm's man page (file pyp2rpm.1 in the main project directory) should be updated as well. It should be possible to generate man page from the help message directly, but I've never tried how it works https://www.gnu.org/software/help2man/
Updated.

mcyprian commented:
Could you please also update SYNOPSIS and AUTHOR sections of the man page? The rest of the updates look good to me :)

irushchyshyn committed:
Update SYNOPSIS and AUTHOR sections of the man page

irushchyshyn commented:
Updated in eaeed79.
Sorry for that, did not notice that it was also changed.
